source/MaterialXCore/Value.cpp

```diff
@@ -94,7 +95,7 @@ template <class T> void stringToData(const string& str, enable_if_std_vector_t<T
 template <class T> void dataToString(const T& data, string& str)
 {
     std::stringstream ss;
-
+    ss.imbue(std::locale::classic());
```

Comment:
Make sure to preserve the newline here, after `ss.imbue` and before the code block that handles float formatting and precision.